internal/services/containers/container_group_data_source.go

```diff
@@ -44,6 +45,17 @@ func dataSourceContainerGroup() *pluginsdk.Resource {
 				Computed: true,
 			},
 
+			"subnets": {
```

Comment:
this should be subnet_ids?
```suggestion
			"subnet_ids": {
```

Reply:
@katbyte Ah, good catch. I was deciding in the very beginning what to select `subnets` or `subnet_ids` (as sometime `subnets` is used for same IDs members). 
I have added `identity` support into data source and added a test to check `identity` and `subnet_ids` in data source.